### PR TITLE
GameList: Show (Disc 1) for first disc of two-disc games

### PR DESCRIPTION
**Pull request by Dentomologist**
Append (Disc 1) label to the first disc of two-disc games, rather than only labelling the second disc.

I don't own any two-disc games so my testing was necessarily hacky; if someone with a larger collection can test it with some real two-disc games that'll be helpful.

I got the list of game ids from the wiki, so hopefully that template is complete.

### CODE REVIEW

File: Source/Core/UICommon/GameFile.cpp [hunk 1]
```diff
@@ -547,6 +549,106 @@ std::vector<DiscIO::Language> GameFile::GetLanguages() const
...
+      // Tiger Woods PGA Tour 2005
+      "G5TP69",
+      "G5TE69",
+      // Tom Clancy's Splinter Cell: Chaos Theory
+      "GCJE41",
+      "GCJP41",
+      // Tom Clancy's Splinter Cell: Double Agent (GC)
+      "GWYX41",
+      "GWYE41",
+  };
```
Comment (jordan-woyak) [inline Source/Core/UICommon/GameFile.cpp:647]: Would it make sense to instead have of an array of the first 3-letter codes and match against just the first 3 letters to shorten this list? Or is that not perfectly reliable? I'm not 100% sure on how the game-IDs are structured.

File: Source/Core/UICommon/GameFile.cpp [hunk 2]
```diff
@@ -547,6 +549,106 @@ std::vector<DiscIO::Language> GameFile::GetLanguages() const
...
+  return std::find(two_disc_game_ids.begin(), two_disc_game_ids.end(), game_id) !=
+         two_disc_game_ids.end();
```
Comment (jordan-woyak) [inline Source/Core/UICommon/GameFile.cpp:649]: If the list was sorted you could replace this with `std::binary_search`. Not that this is a particularly hot-path of code, but it's an easy optimization. :P

File: Source/Core/UICommon/GameFile.cpp (hunk 1 above)
Comment (Tilka) [reply Source/Core/UICommon/GameFile.cpp:647]: The first three characters are enough. See https://wiki.dolphin-emu.org/index.php?title=GameIDs

File: Source/Core/Core/State.cpp [hunk 3]
```diff
@@ -98,7 +98,7 @@ static size_t s_state_writes_in_queue;
 static std::condition_variable s_state_write_queue_is_empty;
 
 // Don't forget to increase this after doing changes on the savestate system
-constexpr u32 STATE_VERSION = 167;  // Last changed in PR 12494
+constexpr u32 STATE_VERSION = 168;  // Last changed in PR 12702
```
Comment (JosJuice) [inline Source/Core/Core/State.cpp:101]: I don't think you need to change this. The DoState in GameFile.cpp is linked to CACHE_REVISION, not STATE_VERSION.

**Comment by Dentomologist** [comment]
I've incorporated the suggestions above:
* Use game_id prefixes instead of the full ID.
* Sort the game_id prefix list and use std::binary_search.
* Retain check for disc number > 1 in case unofficial games have multiple discs or similar.
* Revert the STATE_VERSION change.

**Review by jordan-woyak — APPROVED** [review]
LGTM. untested.

**Review by dreamsyntax — APPROVED** [review]

**Pull request merged**